base/linalg/triangular.jl

```diff
@@ -1900,14 +1907,10 @@ end
 function sqrtm{T}(A::UnitUpperTriangular{T})
```

Comment:
After improving the type stability, the specialised implementation for `UnitUpperTriangular` is faster than converting.
Now, on 0.5.0 using `@benchmark`ing, I see a *slowdown* when unwrapping the type first in `sqrtm`. Relatedly, on my machine (juliabox.com) there appears to be a slowdown for the atomic operations:

    @benchmark $A[1,2]
      median time:      2.650 ns
    @benchmark $T[1,2]
      median time:      2.223 ns

(what does the `$` do here?)
This persists when wrapping these calls in a function. All in all I am hesitant to make the unwrapping change for `sqrtm` (but not against)